Mathlib/Data/Finset/Sups.lean

```diff
@@ -32,18 +32,28 @@ We define the following notation in locale `FinsetFamily`:
 [B. Bollobás, *Combinatorics*][bollobas1986]
 -/
 
+-- TODO: Is there a better spot for those two instances?
+namespace Finset
+variable {α : Type*} [Preorder α] [@DecidableRel α (· ≤ ·)] {s : Finset α}
+
+instance decidablePredMemUpperClosure : DecidablePred (· ∈ upperClosure (s : Set α)) :=
+  fun _ ↦ decidableExistsAndFinset
+
+instance decidablePredMemLowerClosure : DecidablePred (· ∈ lowerClosure (s : Set α)) :=
+  fun _ ↦ decidableExistsAndFinset
+
+end Finset
```

Comment:
Yes, you could add a general `DecidablePred (· ∈ lowerClosure)` instance that assumes the unfolded statement is decidable, right next to the definition of `lowerClosure`.